pkg/services/ngalert/metrics/multi_org_alertmanager.go

```diff
@@ -81,12 +97,66 @@ type AlertmanagerAggregatedMetrics struct {
 	silencesQueryDuration           *prometheus.Desc
 	silences                        *prometheus.Desc
 	silencesPropagatedMessagesTotal *prometheus.Desc
+
+	// exported metrics, gathered from Alertmanager Dispatcher
+	dispatchAggrGroups         *prometheus.Desc
+	dispatchProcessingDuration *prometheus.Desc
 }
 
 func NewAlertmanagerAggregatedMetrics(registries *metrics.TenantRegistries) *AlertmanagerAggregatedMetrics {
 	aggregatedMetrics := &AlertmanagerAggregatedMetrics{
 		registries: registries,
 
+		numNotifications: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_notifications_total", Namespace, Subsystem),
+			"The total number of attempted notifications.",
+			[]string{"org", "integration"}, nil),
+		numFailedNotifications: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_notifications_failed_total", Namespace, Subsystem),
+			"The total number of failed notifications.",
+			[]string{"org", "integration"}, nil),
+		numNotificationRequestsTotal: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_notification_requests_total", Namespace, Subsystem),
+			"The total number of attempted notification requests.",
+			[]string{"org", "integration"}, nil),
+		numNotificationRequestsFailedTotal: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_notification_requests_failed_total", Namespace, Subsystem),
+			"The total number of failed notification requests.",
+			[]string{"org", "integration"}, nil),
+		notificationLatencySeconds: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_notification_latency_seconds", Namespace, Subsystem),
+			"The latency of notifications in seconds.",
+			nil, nil),
+
+		nflogGCDuration: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_nflog_gc_duration_seconds", Namespace, Subsystem),
+			"Duration of the last notification log garbage collection cycle.",
+			nil, nil),
+		nflogSnapshotDuration: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_nflog_snapshot_duration_seconds", Namespace, Subsystem),
+			"Duration of the last notification log snapshot.",
+			nil, nil),
+		nflogSnapshotSize: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_nflog_snapshot_size_bytes", Namespace, Subsystem),
+			"Size of the last notification log snapshot in bytes.",
+			nil, nil),
+		nflogQueriesTotal: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_nflog_queries_total", Namespace, Subsystem),
+			"Number of notification log queries were received.",
+			nil, nil),
+		nflogQueryErrorsTotal: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_nflog_query_errors_total", Namespace, Subsystem),
+			"Number notification log received queries that failed.",
+			nil, nil),
+		nflogQueryDuration: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_nflog_query_duration_seconds", Namespace, Subsystem),
+			"Duration of notification log query evaluation.",
+			nil, nil),
+		nflogPropagatedMessagesTotal: prometheus.NewDesc(
+			fmt.Sprintf("%s_%s_nflog_gossip_messages_propagated_total", Namespace, Subsystem),
+			"Number of received gossip messages that have been further gossiped.",
+			nil, nil),
```

Comment:
For reviewers: [Upstream nflog metrics](https://github.com/prometheus/alertmanager/blob/main/nflog/nflog.go#L106)